src/sentry/static/sentry/app/views/alerts/rules/details/metricChart.tsx

```diff
@@ -248,4 +280,253 @@ export default class MetricChart extends React.PureComponent<Props, State> {
+          }
+
+          const series: LineChartSeries[] = [...timeseriesData];
+          // Ensure series data appears above incident lines
+          series[0].z = 100;
+          const dataArr = timeseriesData[0].data;
+          const maxSeriesValue = dataArr.reduce(
+            (currMax, coord) => Math.max(currMax, coord.value),
+            0
+          );
+          const firstPoint = Number(dataArr[0].name);
+          const lastPoint = dataArr[dataArr.length - 1].name as number;
+          const totalDuration = lastPoint - firstPoint;
+          let criticalDuration = 0;
+          let warningDuration = 0;
+          const resolvedArea = {
+            seriesName: 'Resolved Area',
+            type: 'line',
+            markLine: MarkLine({
+              silent: true,
+              lineStyle: {color: theme.green300, type: 'solid', width: 4},
+              data: [[{coord: [firstPoint, 0]}, {coord: [lastPoint, 0]}] as any],
+            }),
+            data: [],
+          };
+          series.push(resolvedArea);
+          if (incidents) {
+            // select incidents that fall within the graph range
+            const periodStart = moment.utc(firstPoint);
+
+            incidents
+              .filter(
+                incident =>
+                  !incident.dateClosed || moment(incident.dateClosed).isAfter(periodStart)
+              )
+              .forEach(incident => {
+                const statusChanges = incident.activities
+                  ?.filter(
+                    ({type, value}) =>
+                      type === IncidentActivityType.STATUS_CHANGE &&
+                      value &&
+                      [
+                        `${IncidentStatus.WARNING}`,
+                        `${IncidentStatus.CRITICAL}`,
+                      ].includes(value)
+                  )
+                  .sort(
+                    (a, b) =>
+                      moment(a.dateCreated).valueOf() - moment(b.dateCreated).valueOf()
+                  );
+
+                const incidentEnd = incident.dateClosed ?? moment().valueOf();
+
+                const timeWindowMs = rule.timeWindow * 60 * 1000;
+
+                series.push(
+                  createIncidentSeries(
+                    rule.triggers.find(({label}) => label === 'warning') &&
+                      statusChanges &&
+                      !statusChanges.find(
+                        ({value}) => value === `${IncidentStatus.CRITICAL}`
+                      )
+                      ? theme.yellow300
+                      : theme.red300,
+                    moment(incident.dateStarted).valueOf(),
+                    incident.identifier
+                  )
+                );
+                const areaStart = moment(incident.dateStarted).valueOf();
+                const areaEnd =
+                  statusChanges?.length && statusChanges[0].dateCreated
+                    ? moment(statusChanges[0].dateCreated).valueOf() - timeWindowMs
```

Comment:
just making sure I understand the status area painting. This first area here is to paint a guaranteed section regardless of status areas, and then the for loop below is to paint the rest of the status changes?

If so, you may be able to cheat a little bit and paint one large area (either red or yellow) under the incident and then use the for loop to paint over it. This is what I did with the resolved area. I'm painting a green area which spans the whole timeline on the chart and then putting red status areas over it. I imagine you could do something similar here just to avoid having to do these calculations with looking into whether a statusChange[0] exists.

Reply:
That would probably result in a similar result but I think it'd be more complicated